src/App.tsx

```diff
@@ -14,11 +14,106 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState(goodsFromServer);
+  const [visibleList, setVisibleList] = useState(false);
 
-export default App;
+  const isVisibleList = () => {
+    setVisibleList(!visibleList);
+  };
+
+  const makeReverse = () => {
+    setGoods([...goods].reverse());
+  };
+
+  const makeSortAlphabet = () => {
+    setGoods([...goods].sort((a, b) => a.localeCompare(b)));
+  };
+
+  const makeSortLength = () => {
+    setGoods([...goods].sort((a, b) => a.length - b.length));
+  };
+
+  const makeReset = () => {
+    setGoods(goodsFromServer);
+  };
+
+  return (
+    <div className="App">
+      <h1 className="title">Goods</h1>
+
+      {!visibleList && (
+        <button
```

Comment:
```suggestion
      {!visibleList ? ( ) : ( )}
        
```
In this case maybe it will better to use  ternary operator. I think  it can little bit improve the readability of the code.